envi/archs/rx/disasm.py

```diff
@@ -0,0 +1,1349 @@
+        mcanv.addText(" ")
+
+        # Allow each of our operands to render
+        imax = len(self.opers)
+        lasti = imax - 1
+        for i in range(imax):
+            oper = self.opers[i]
+            oper.render(mcanv, self, i)
+            if i != lasti:
+                mcanv.addText(", ")
+
+    def getBranches(self, emu=None):
+        ret = []
+
+        # To start with we have no flags ( except our arch )
+        flags = self.iflags & envi.ARCH_MASK
+        addb = False
+
+        # If we are a conditional branch, even our fallthrough
+        # case is conditional...
+        if (self.iflags & IF_BRANCH):
+            addb = True
+            if not (self.iflags & IF_NOFALL):
+                flags |= envi.BR_COND
+
+        # If we can fall through, reflect that...
+        if not self.iflags & envi.IF_NOFALL:
+            ret.append((self.va + self.size, flags|envi.BR_FALL))
+
+        # if we have no operands, it has no further branches...
+        if len(self.opers) == 0:
+            return ret
+
+        # Check for a call...
+        if self.iflags & IF_CALL:
+            flags |= envi.BR_PROC
+            addb = True
+
+        if addb:
+            oper0 = self.opers[0]
+            if oper0.isDeref():
+                flags |= envi.BR_DEREF
+                tova = oper0.getOperAddr(self, emu=emu)
+            else:
+                tova = oper0.getOperValue(self, emu=emu)
+
+            ret.append((tova, flags))
+
+        return ret
+
+#FIXME: This should just be wrapped into envi
+def addrToName(mcanv, va):
+    sym = mcanv.syms.getSymByAddr(va)
+    if sym is not None:
+        return repr(sym)
+    return "0x%.4x" % va
+
+def renderPossibleLocation(mcanv, op, idx, value):
+    hint = mcanv.syms.getSymHint(op.va, idx)
+    if hint is not None:
+        if mcanv.mem.isValidPointer(value):
+            mcanv.addVaText(hint, value)
+        else:
+            mcanv.addNameText(hint)
+    elif mcanv.mem.isValidPointer(value):
+        name = addrToName(mcanv, value)
+        mcanv.addVaText(name, value)
+    else:
+        value &= 0xffff
+        mcanv.addNameText('0x%x' % value)
+
+class RxRegDirectOper(envi.RegisterOper):
+    '''
+    register operand
+    '''
+
+    def __init__(self, reg, va=0, oflags=0):
+        if reg is None:
+            raise envi.InvalidInstruction(mesg="None Reg Type!",
+                    bytez='f00!', va=va)
+        self.va = va
+        self.reg = reg
+        self.oflags = oflags
+
+    def __eq__(self, oper):
+        if not isinstance(oper, self.__class__):
+            return False
+        if self.reg != oper.reg:
+            return False
+        if self.oflags != oper.oflags:
+            return False
+        return True
+    
+    def involvesPC(self):
+        return self.reg == regs.REG_PC
+
+    def getWidth(self):
+        return regs.rctx.getRegisterWidth(self.reg) // 8
+
+    def getOperValue(self, op, emu=None, codeflow=False):
```

Comment:
I may have asked this in the ARM PR, but does anybody call this with `codeflow=True`? I can't find anybody who does.